### PR TITLE
Simplifies ScheduleResults.job_verdicts

### PR DESCRIPTION
**Pull request by ruomingp**
... so that its entries are ordered by the global priorities.

This allows the caller to export the job priorities and eventually surface the information on the Bastion UI.

### CODE REVIEW

**Review by markblee — APPROVED** [review]

**Comment by ruomingp** [comment]
Thanks! PTAL as I added some comments?

**Review by WeersProductions — COMMENTED** [review]
Thanks a lot! These comments are super helpful.
Left two nits on comments, non-blocking.

File: axlearn/cloud/common/scheduler_test.py [hunk 1]
```diff
@@ -294,10 +301,12 @@ def test_validate_tiers(self):
             },
             expected_project_limits={"a": {"v4": 7}, "b": {"v4": 8}},
             expected_verdicts={"a1": True, "b1": True, "b2": True},
-            expected_tiers={"a1": 0, "b1": 0, "b2": 1},
+            # "b1" is ahead of "a1" since it requests less resource.
```
[the remark this answers is not in the document]
Comment (ruomingp) [reply axlearn/cloud/common/scheduler_test.py]: Done.

**Comment by ruomingp** [comment]
Thanks. PTAL?

**Review by WeersProductions — APPROVED** [review]
Thanks!

**Pull request merged**